src/browser/components/MainPage.jsx

```diff
@@ -618,7 +626,7 @@ export default class MainPage extends React.Component {
         activeKey={this.state.key}
         onSelect={this.handleSelect}
         onAddServer={this.addServer}
-        showAddServerButton={this.props.showAddServerButton}
+        showAddServerButton={this.state.showSettingsModal ? false : this.props.showAddServerButton}
```

Comment:
Added to maintain desired UI feature from https://github.com/mattermost/desktop/pull/1174